dataset-processor-python/data_loader.py

```diff
@@ -31,25 +43,186 @@ class DatasetWriter(DataLoader):
+        return re.match(pattern, version) is not None
+
+    def process_data(self):
         """
-        get the inputs from the user
+        Process each row of the csv file and compile the json files
         """
-        # version is required, title and description are optional, default values are set as empty string
-        # return version, title, description
-        pass
+        for row in self.data:
+            file_info, features = self.get_row_data(row)
+            self.compile_cell_feature_analysis(file_info, features)
+        self.compile_feature_defs()
+        self.compile_dataset()
 
-    def create_dataset_folder(self):
+    def get_row_data(self, row):
+        """
+        Separates the file info and features from the row
+        """
+        file_info, features = [], []
+        for column, value in row.items():
+            value = self.convert_str_to_num(value)
+            if column in self.FILEINFO_COLUMN_NAMES:
+                file_info.append(value)
+            else:
+                self.process_features(column, value, features)
+        return file_info, features
+
+    def convert_str_to_num(self, value):
+        try:
+            numeric_value = float(value)
+            if numeric_value.is_integer():
+                return int(numeric_value)
+            return numeric_value
+        except ValueError:
+            # if not number, return the original value
+            return value
+
+    def process_features(self, column, value, features):
+        if self.is_valid_feature_value(value):
+            features.append(value)
+            self.update_feature_metadata(column, value)
+        else:
+            print(f"Invalid value: {value} in column {column}")
+            return
+
+    def is_valid_feature_value(self, value):
+        return pd.isna(value) or isinstance(value, (int, float))
+
+    def update_feature_metadata(self, column, value):
+        is_discrete = self.is_discrete(value)
+        self.discrete_features_dict[column] = is_discrete
+        if column not in self.feature_def_keys:
+            self.feature_def_keys.append(column)
+
+    def is_discrete(self, numeric_value):
         """
-        create a folder with the database_name
-        initialize the folder with required json files
+        Determines if the numeric value is discrete
         """
-        # folder_name = self.dataset_name + "_" + version
-        # path = os.path.join("data", folder_name)
-        # os.makedirs(path, exist_ok=True)
-        # create json files -> feature_defs.json, dataset.json, cell_feature_analysis.json, image_settings.json
-        pass
+        return not pd.isna(numeric_value) and numeric_value == int(numeric_value)
+
+    def compile_cell_feature_analysis(self, file_info, features):
+        self.cell_feature_analysis_file.append(
+            {"file_info": file_info, "features": features}
+        )
+
+    def compile_feature_defs(self):
+        description = ""
+        tooltip = ""
```

Comment:
Will these be defined by the source file?

Reply:
The user can either define these settings through input or directly modify the json files as needed. Our nest step will be to set up the required settings.